test/ref_ops_test.cpp

```diff
@@ -6458,6 +6458,168 @@ TEST_CASE(quantizelinear)
     }
 }
 
+TEST_CASE(random_uniform_test)
+{
+    migraphx::program p;
+    auto* mm = p.get_main_module();
+    uint64_t seed(0);
+    size_t sample_size(200);
+
+    //      Shape of the random data
+    migraphx::shape rs{migraphx::shape::float_type, {1, sample_size}};
+
+    // data tensor must be allocated at this point but does not need to be initialized.
+    std::vector<float> data(sample_size);
+    auto input = mm->add_literal(migraphx::literal(rs, data));
+
+    // Runtime randomization seed
+    migraphx::shape seed_shape{migraphx::shape::uint64_type, {1}};
+    std::vector<uint64_t> seed_data{seed};
+    auto seed_input = mm->add_literal(migraphx::literal(seed_shape, seed_data));
+
+    mm->add_instruction(migraphx::make_op("random_uniform"), seed_input, input);
+    p.compile(migraphx::make_target("ref"));
+
+    migraphx::parameter_map params0;
+    auto result = p.eval(params0).back();
+    std::vector<float> result_vec(sample_size);
+    result.visit([&](auto output) { result_vec.assign(output.begin(), output.end()); });
+
+    // Compare result with the STL's mt19937 generator
+    std::mt19937 gen(seed);
+    std::uniform_real_distribution<> dis(0.0, 1.0);
+    std::vector<float> rand_samples(sample_size);
+    std::generate(rand_samples.begin(), rand_samples.end(), [&]() { return dis(gen); });
+    EXPECT(migraphx::verify::verify_range(result_vec, rand_samples, 100000));
```

Comment:
Why are these tolerance values so high? Aren't these using the same random number generator and seed?

Reply:
The actual error should be exactly 0, but I'm leery of ever comparing floating-point values without an error epsilon.  The tolerance given is not really so high since verify_range() multiplies it by an epsilon, but I'm going to reduce it to 100 anyway.